### ISSUE TITLE
Wrong type reported for associative array assignment (with sprintf func result)

### ISSUE THREAD

**Issue by pmaselkowski** [scrutinizer-ci/php-analyzer#725]
Variable is of type `string[]` and it's declared like that in class field.

However analyzer reports it as a bug when assigning like this:
```php
$this->bindings = [
			'data-bind' => sprintf('enumFormatter: {data: +%s, values: %s}', $this->name, JavaScript::encode($values))
		];
```

Detected type is weird, as it contains expression which looks somewhat like a json encoded value of this array:
```
array<string,string,{"data-bind":"strin{"data-bind":"string"}>
```
see https://scrutinizer-ci.com/g/Maslosoft/IlmatarWidgets/inspections/63adadcc-42bd-4219-88d9-ed10e02ffaa3/issues/files/src/Grid/Column/Bool.php?status=new&orderField=path&order=asc&issueId=454720a8-7b9e-49a4-9726-9b1df66dbdde
